File: src/Game/Systems/TestSystem.lua
local System = require("Game/System")
local Timer = require("Timer")
local Vector2 = require("Math/Vector2")
local EventCenter = require("Game/EventCenter")

local TestSystem = class("TestSystem", System)

function TestSystem:OnClickedZ()
    
end

function TestSystem:Awake()
    EventCenter.Keyboard:AddListener(self.OnClickedZ, self)
end

function TestSystem:Start()
    log("Test system is run")
end

function TestSystem:Update(dt)
end

return TestSystem

File: src/Game/Systems/TestSystem.ts
import { Logger } from "../../Logger";
import { Vector2 } from "../../Math/Vector2";
import { Timer } from "../../Timer";
import { Keyboard } from "../EventCenter";
import { ISystem } from "../System";

export class TestSystem implements ISystem {

    public OnClickedZ(data: { player: player, key: oskeytype } | void): void {
        //
    }

    public Awake(): void {
        Keyboard.AddListener(this, this.OnClickedZ);
    }

    public Start(): void {
        Logger.Log("Test system is run");
    }

    public Update(dt: float): void {
        //
    }

}
